SwiftCompilerSources/Sources/Optimizer/Utilities/OptUtils.swift

```diff
@@ -546,6 +546,51 @@ extension Instruction {
     // Eventually we want to replace the SILCombine implementation with this one.
     return nil
   }
+
+  /// Returns true if a destroy of `type` must not be moved across this instruction.
+  func isBarrierForDestroy(of type: Type, _ context: some Context) -> Bool {
+    let instEffects = memoryEffects
+    if instEffects == .noEffects || type.isTrivial(in: parentFunction) {
+      return false
+    }
+    if type.isMoveOnly {
```

Comment:
As a matter of personal style, did you consider this?
```
guard type.isMoveOnly else {
   return isDeinitBarrier(context.calleeAnalysis)
}
// move-only implementation
```
Or just factoring the moveOnly logic?